core/src/main/scala/zio/temporal/workflow/ZSaga.scala

```diff
@@ -215,21 +215,21 @@ object ZSaga {
 
     def interpret[A0](saga: ZSaga[A0]): Either[Throwable, A0] =
       saga match {
-        case succeed: Succeed[A0] => Right(succeed.value)
```

Comment:
Was emitting a warning as the program is not able to type check the `A0` type as it's eliminated at runtime